ush/soca/marine_recenter.py

```diff
@@ -0,0 +1,248 @@
+#!/usr/bin/env python3
+
+from datetime import datetime, timedelta
+import f90nml
+from logging import getLogger
+import os
+from soca import bkg_utils
+from typing import Dict
+import ufsda
+from ufsda.stage import soca_fix
+from wxflow import (chdir,
+                    Executable,
+                    FileHandler,
+                    logit,
+                    Task,
+                    Template,
+                    TemplateConstants,
+                    WorkflowException,
+                    YAMLFile)
+
+logger = getLogger(__name__.split('.')[-1])
+
+
+class MarineRecenter(Task):
+    """
+    Class for global ocean analysis recentering task
+    """
+
+    @logit(logger, name="MarineRecenter")
+    def __init__(self, config: Dict) -> None:
+        """Constructor for ocean recentering task
+        Parameters:
+        ------------
+        config: Dict
+            configuration, namely evironment variables
+        Returns:
+        --------
+        None
+        """
+
+        logger.info("init")
+        super().__init__(config)
+
+        # Variables of convenience
+        # TODO (AFE) maybe the g- vars should be done in the jjob
+        PDY = self.runtime_config['PDY']
+        cyc = self.runtime_config['cyc']
+        cdate = PDY + timedelta(hours=cyc)
+        gdate = cdate - timedelta(hours=6)
+        self.runtime_config['gcyc'] = gdate.strftime("%H")
+        self.runtime_config['gPDY'] = datetime(gdate.year,
+                                               gdate.month,
+                                               gdate.day)
+
+        gdas_home = os.path.join(config['HOMEgfs'], 'sorc', 'gdas.cd')
+
+        half_assim_freq = timedelta(hours=int(config['assim_freq'])/2)
+        window_begin = cdate - half_assim_freq
+        window_begin_iso = window_begin.strftime('%Y-%m-%dT%H:%M:%SZ')
+        window_middle_iso = cdate.strftime('%Y-%m-%dT%H:%M:%SZ')
+
+        self.window_config = {'window_begin': f"{window_begin.strftime('%Y-%m-%dT%H:%M:%SZ')}",
+                              'ATM_WINDOW_BEGIN': window_begin_iso,
+                              'ATM_WINDOW_MIDDLE': window_middle_iso,
+                              'ENS_SIZE': self.config.NMEM_ENS,
```

Comment:
yeah like I said above, we shouldn't rename variables, and it should be NMEM_ENS everywhere.

Reply:
changed